macosx/FileNameCell.mm

```diff
@@ -20,45 +20,51 @@
 static CGFloat const kPaddingBetweenNameAndFolderStatus = 4.0;
 static CGFloat const kPaddingExpansionFrame = 2.0;
 
-@interface FileNameCell ()
-
-@property(nonatomic, readonly) NSAttributedString* attributedTitle;
-@property(nonatomic, readonly) NSAttributedString* attributedStatus;
-@property(nonatomic, readonly) NSMutableDictionary* fTitleAttributes;
-@property(nonatomic, readonly) NSMutableDictionary* fStatusAttributes;
-
-@end
-
-@implementation FileNameCell
-
-- (instancetype)init
+static NSParagraphStyle* kParagraphStyle()
 {
-    if ((self = [super init]))
-    {
-        NSMutableParagraphStyle* paragraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
-        paragraphStyle.lineBreakMode = NSLineBreakByTruncatingMiddle;
-
-        _fTitleAttributes = [[NSMutableDictionary alloc]
-            initWithObjectsAndKeys:[NSFont messageFontOfSize:12.0], NSFontAttributeName, paragraphStyle, NSParagraphStyleAttributeName, nil];
-
-        NSMutableParagraphStyle* statusParagraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
-        statusParagraphStyle.lineBreakMode = NSLineBreakByTruncatingTail;
-
-        _fStatusAttributes = [[NSMutableDictionary alloc]
-            initWithObjectsAndKeys:[NSFont messageFontOfSize:9.0], NSFontAttributeName, statusParagraphStyle, NSParagraphStyleAttributeName, nil];
-    }
-    return self;
+    NSMutableParagraphStyle* paragraphStyle = [NSParagraphStyle.defaultParagraphStyle mutableCopy];
+    paragraphStyle.lineBreakMode = NSLineBreakByTruncatingMiddle;
+    return paragraphStyle;
```

Comment:
Strictly speaking this should be `return [paragraphStyle copy];` – here and below. Otherwise the real type leaks to the runtime, which can lead to issues

Reply:
Doesn't matter: it's not a public API. We don't need a copy, so I changed the signature to NSMutableParagraphStyle.